src/components/post-viewer/post-viewer.tsx

```diff
@@ -40,119 +50,129 @@ export const PostViewer: FC<Props> = ({ postRef }) => {
     setZoom(1);
   };
 
+  const { bg: contextBg, text: contextText } = colorItemVariants[posterBg];
+
   return (
-    <div className="relative col-span-2 flex max-w-full bg-slate-800 text-white md:max-h-screen">
-      <IconContext.Provider value={{ className: "text-white" }}>
-        <div className="absolute right-5 top-5 z-10 flex flex-col items-center gap-2 bg-slate-900 px-2 text-xs opacity-30 transition-opacity hover:opacity-100">
-          <button
-            className="cursor-pointer p-1 text-base"
-            onClick={handleZoom.bind(null, 0.05)}
-          >
-            <MdAdd />
-          </button>
-          <p className="text-sm">{zoom.toFixed(2)}</p>
-          <button
-            className="cursor-pointer p-1"
-            onClick={handleZoom.bind(null, -0.05)}
-          >
-            <MdRemove />
-          </button>
-          <button className="py-1 text-base" onClick={fitToViewer}>
-            <MdFitScreen />
-          </button>
-          <button className="py-1 text-base" onClick={resetZoom}>
-            <GrUndo />
-          </button>
-        </div>
-      </IconContext.Provider>
-      <div
-        ref={viewerRef}
-        className="flex max-h-full max-w-full overflow-scroll p-1"
-      >
-        <div
-          className="transition-transform"
-          style={{ transform: `scale(${zoom})`, transformOrigin: "top left" }}
-        >
+    <div className="relative col-span-3 max-h-96 max-w-full bg-slate-800 text-white md:max-h-screen">
+      <ViewerControls
+        zoom={zoom}
+        handleZoom={handleZoom}
+        fitToViewer={fitToViewer}
+        resetZoom={resetZoom}
+      />
+      <div ref={viewerRef} className="flex h-full p-1 md:h-full md:w-full">
+        <ScrollArea>
           <div
-            ref={postRef}
-            className="relative grid aspect-square w-[1080px] grid-cols-3 grid-rows-4 gap-2 bg-primary px-8 py-24 text-3xl"
+            className="transition-transform"
+            style={{ transform: `scale(${zoom})`, transformOrigin: "top left" }}
           >
-            <span className="absolute left-8 top-8 flex flex-row gap-2 text-5xl">
-              <h1>CSI PRO</h1>
-              {posterForm && posterForm?.watch("event")?.length > 0 && (
-                <NameEmphasis>{posterForm?.watch("event")}</NameEmphasis>
+            <div
+              ref={postRef}
+              className={cn(
+                "relative grid aspect-square w-[1080px] grid-cols-3 grid-rows-4 gap-2 px-8 py-24 text-3xl transition-all",
```

Comment:
what this part of the code does exactly?

Reply:
This part of the code is a div that takes in a ref, and this div represents the image template. A ref (reference) is an object created to manipulate the DOM that, in this case, we use to be able to get the current state of this div and its contents from another component, and then download it. Lastly, className takes in all of the styles that apply to this particular div.

You can read about refs [here](https://react.dev/reference/react/useRef).